### PR TITLE
WIP: Add new test case OCS-371

### PR DESCRIPTION
**Pull request by ebenahar**
Delete PVC while IO is in progress

### CODE REVIEW

File: templates/ocs-deployment/CephBlockPool.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: {{ cephblockpool_name | default('ocsci-cephblockpool') }}
+  namespace: {{ cluster_namespace | default('openshift-storage') }}
```
Comment (keesturam) [inline templates/ocs-deployment/CephBlockPool.yaml]: I realize that the namespace for block pool cannot be anything other than openshift-storage. Rook looks to create this block pool in the ceph cluster in openshift-storage namespace. So if one provides any other namespace it simply fails.

  Warning  ProvisioningFailed  10s (x2 over 25s)  ceph.rook.io/block_rook-ceph-operator-5b6856f864-t67gx_18bb8827-7d15-11e9-83b8-0a580a800217  failed to provision volume with StorageClass "ocs-qe": Failed to create rook block image krk-rbd/pvc-3a4ea058-7df8-11e9-9bd6-0ac430f8eef8: failed to create image pvc-3a4ea058-7df8-11e9-9bd6-0ac430f8eef8 in pool krk-rbd of size 4294967296: Failed to complete '': exit status 1. global_init: unable to open config file from search list /var/lib/rook/krk/krk.config

please note that krk is the ceph cluster rook is trying to find and fail.

So, this value should be hardcoded with 'openshift-storage'. I'll also send a patch to replace that in the storage-manigest.yaml file.

File: templates/ocs-deployment/CephBlockPool.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+spec:
+  failureDomain: {{ failureDomain | default('host') }}
+  replicated:
+    size: {{ size | default('2') }}
```
Comment (keesturam) [inline templates/ocs-deployment/CephBlockPool.yaml]: Let us have the default replica size as '3' instead of '2' as that is what will be supported.

File: templates/ocs-deployment/CephBlockPool.yaml (hunk 1 above)
Comment (ebenahar) [reply templates/ocs-deployment/CephBlockPool.yaml]: @keesturam, in that case, I suggest to leave it as it is with openshift-storage as the default.

File: templates/ocs-deployment/CephBlockPool.yaml (hunk 2 above)
Comment (ebenahar) [reply templates/ocs-deployment/CephBlockPool.yaml]: Done

File: tests/helpers.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import logging
+import ocs.ocp
+import ocs.defaults as defaults
+from utility import templating
+
+logger = logging.getLogger(__name__)
+
+
+OCP = ocs.ocp.OCP(
+    kind='Service', namespace=defaults.ROOK_CLUSTER_NAMESPACE
```
Comment (keesturam) [inline tests/helpers.py]: namespace in this case is defined in two places. 

1) namespace=defaults.ROOK_CLUSTER_NAMESPACE
2) PROJECT_NAME = 'test-project'

I think this is be problematic. I'd not define the namespace here, rather define it like how you have done in line 22 and use that value throughout the test. Also, by doing that we use the same OCP object for different namespaces if required. This gives more flexibility while writing test case instead of going with a pre-defined name in config file. 

p.s., This comment is applicable for all the places where namespace is defined in ocs.ocp.OCP object.

File: templates/ocs-deployment/StorageClass.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+kind: StorageClass
+metadata:
+  name: {{ sc_name | default('ocsci-storageclass') }}
+provisioner: ceph.rook.io/block
+parameters:
+  blockPool: ocsci-cephblockpool
+  # Specify the namespace of the rook cluster from which to create volumes.
+  # If not specified, it will use `rook` as the default namespace of the cluster.
+  # This is also the namespace where the cluster will be
+  clusterNamespace: {{ cluster_namespace | default('openshift-storage') }}
```
Comment (keesturam) [inline templates/ocs-deployment/StorageClass.yaml]: This should always be `openshift-storage` as `clusterNamespace` defines where ceph cluster is running.

File: templates/ocs-deployment/CephBlockPool.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+---
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: {{ cephblockpool_name | default('ocsci-cephblockpool') }}
+  namespace: {{ cluster_namespace | default('openshift-storage') }}
+spec:
+  failureDomain: {{ failureDomain | default('host') }}
+  replicated:
+    size: {{ size | default('3') }}
```
Comment (RazTamir) [inline templates/ocs-deployment/CephBlockPool.yaml]: it will work with just '3'?
I think it should be '3Gi'

File: templates/ocs-deployment/Pod.yaml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  name: {{ pod_name | default('ocsci-pod') }}
+spec:
+  containers:
+   - name: web-server
+     image: nginx
+     volumeMounts:
+       - name: mypvc
+         mountPath: /var/lib/www/html
+  volumes:
+   - name: mypvc
```
Comment (RazTamir) [inline templates/ocs-deployment/Pod.yaml]: jinja2

File: templates/ocs-deployment/Pod.yaml [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+     persistentVolumeClaim:
+       claimName: ocsci-pvc
```
Comment (RazTamir) [inline templates/ocs-deployment/Pod.yaml]: same

File: templates/ocs-deployment/Pod.yaml [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+apiVersion: v1
+kind: Pod
+metadata:
+  name: {{ pod_name | default('ocsci-pod') }}
+spec:
+  containers:
+   - name: web-server
+     image: nginx
+     volumeMounts:
+       - name: mypvc
```
Comment (RazTamir) [inline templates/ocs-deployment/Pod.yaml]: jinja2

File: tests/fixtures.py [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+
+    """
+    template = os.path.join(TEMPLATES_DIR, "CephBlockPool.yaml")
+    logger.info(f'Creating a Ceph Block Pool')
+
+    templating.dump_to_temp_yaml(template, TEMP_YAML)
+    assert CBP.create(yaml_file=TEMP_YAML)
+    open(TEMP_YAML, 'w').close()
+    # TODO:
+    # wait()
```
Comment (RazTamir) [inline tests/fixtures.py]: you can now implement all the #waits

File: tests/fixtures.py [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+    Create a pod
+
+    """
+    template = os.path.join(TEMPLATES_DIR, "Pod.yaml")
+    logger.info(f'Creating a pod')
+
+    templating.dump_to_temp_yaml(template, TEMP_YAML)
+
+    assert PVC.create(yaml_file=TEMP_YAML)
+    open(TEMP_YAML, 'w').close()
```
Comment (RazTamir) [inline tests/fixtures.py]: you can use delete_file from utils.py

File: tests/fixtures.py [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+    # TODO:
+    # wait()
+
+
+@pytest.fixture()
+def delete_ceph_block_pool(request):
+    """
+    Delete the Ceph block pool
+
```
Comment (RazTamir) [inline tests/fixtures.py]: empty line

File: tests/fixtures.py [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+    templating.dump_to_temp_yaml(template, TEMP_YAML)
+
+    assert PVC.create(yaml_file=TEMP_YAML)
+    open(TEMP_YAML, 'w').close()
+    # TODO:
+    # wait()
+
+
+@pytest.fixture()
+def delete_ceph_block_pool(request):
```
Comment (RazTamir) [inline tests/fixtures.py]: why not add this to the finalizer to create ceph block pool?
Same for all finalizers below

File: tests/helpers.py [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+PROJECT_NAME = 'test-project'
+
+def run_io():
+    """
+    Run IO to a file within the pod
+
+    """
+    OCP.exec_oc_cmd(
+        f"rsh -n openshift-storage ocsci-pod touch /var/lib/www/html/test && "
+        f"dd if=/dev/urandom of=/var/lib/www/html/test bs=1M count=3000 &"
```
Comment (RazTamir) [inline tests/helpers.py]: Can we maybe try to wget some file to mimic a more realistic scenario?
I mean, we don't really sure the urandom is close to real-world IO

File: tests/helpers.py [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+    )
+
+def delete_pvc():
+    """
+
```
Comment (RazTamir) [inline tests/helpers.py]: fix empty lines

File: templates/ocs-deployment/CephBlockPool.yaml (hunk 5 above)
Comment (RazTamir) [reply templates/ocs-deployment/CephBlockPool.yaml]: Ok my mistake.
So please change 'size' to 'replica_count'

File: templates/ocs-deployment/temp.yaml [hunk 15]
```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
```
Comment (RazTamir) [inline templates/ocs-deployment/temp.yaml]: This file should not be part of the PR.
If I understand correctly it is generated by the test and should be deleted as part of the teardown

File: templates/ocs-deployment/StorageClass.yaml [hunk 16]
```diff
@@ -0,0 +1,19 @@
+---
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: {{ sc_name | default('ocsci-storageclass') }}
+provisioner: ceph.rook.io/block
+parameters:
+  blockPool: ocsci-cephblockpool
```
Comment (RazTamir) [inline templates/ocs-deployment/StorageClass.yaml]: jinja2

File: tests/helpers.py (hunk 3 above)
Comment (ebenahar) [reply tests/helpers.py]: Done

File: templates/ocs-deployment/StorageClass.yaml (hunk 4 above)
Comment (ebenahar) [reply templates/ocs-deployment/StorageClass.yaml]: Done

File: tests/fixtures.py [hunk 17]
```diff
@@ -0,0 +1,156 @@
...
+    templating.dump_to_temp_yaml(template, TEMP_YAML)
+
+    assert PVC.create(yaml_file=TEMP_YAML)
+    open(TEMP_YAML, 'w').close()
+    # TODO:
+    # wait()
+
+
+@pytest.fixture()
+def create_pod():
```
[the remark this answers is not in the document]
Comment (ebenahar) [reply tests/fixtures.py]: Thanks, done

File: templates/ocs-deployment/Pod.yaml (hunk 6 above)
Comment (ebenahar) [reply templates/ocs-deployment/Pod.yaml]: done

File: templates/ocs-deployment/Pod.yaml (hunk 7 above)
Comment (ebenahar) [reply templates/ocs-deployment/Pod.yaml]: done

File: templates/ocs-deployment/Pod.yaml (hunk 8 above)
Comment (ebenahar) [reply templates/ocs-deployment/Pod.yaml]: done

File: tests/fixtures.py (hunk 9 above)
Comment (ebenahar) [reply tests/fixtures.py]: done

File: tests/fixtures.py (hunk 10 above)
Comment (ebenahar) [reply tests/fixtures.py]: done

File: tests/fixtures.py (hunk 11 above)
Comment (ebenahar) [reply tests/fixtures.py]: delete

File: tests/fixtures.py (hunk 12 above)
Comment (ebenahar) [reply tests/fixtures.py]: done

File: tests/helpers.py (hunk 13 above)
Comment (ebenahar) [reply tests/helpers.py]: implemented another mechanism with copying a file in while true running in another thread, while deleting a pvc

File: tests/helpers.py (hunk 14 above)
Comment (ebenahar) [reply tests/helpers.py]: done

File: templates/ocs-deployment/temp.yaml (hunk 15 above)
Comment (ebenahar) [reply templates/ocs-deployment/temp.yaml]: Correct :)

File: templates/ocs-deployment/StorageClass.yaml (hunk 16 above)
Comment (ebenahar) [reply templates/ocs-deployment/StorageClass.yaml]: done

File: templates/ocs-deployment/CephBlockPool.yaml (hunk 5 above)
Comment (ebenahar) [reply templates/ocs-deployment/CephBlockPool.yaml]: Forgot about this :) will be fixed in next commit

File: tests/helpers.py [hunk 18]
```diff
@@ -39,6 +44,66 @@ def create_unique_resource_name(resource_description, resource_type):
     return f"{resource_type}_{resource_description[:23]}_{current_date_time[:10]}"
 
 
+def create_resource(
+    resource_kind, resource_name, project_name, yaml_file, **kwargs
```
Comment (RazTamir) [inline tests/helpers.py]: maybe template_yaml instead of yaml_file?

File: suites/manage/ocp/pvc/test_pvc.py [hunk 19]
```diff
@@ -0,0 +1,135 @@
...
+    Create a Ceph block pool
+    """
+
+    def finalizer():
+        """
+        Delete the Ceph block pool created during setup
+        """
+        pass
+
+    pool_body = os.path.join("templates/ocs-deployment", "CephBlockPool.yaml")
```
Comment (vasukulkarni) [inline suites/manage/ocp/pvc/test_pvc.py]: This can be used now and the yaml is loaded in the function: https://github.com/red-hat-storage/ocs-ci/blob/master/ocs/rook.py#L144

File: tests/helpers.py (hunk 18 above)
Comment (ebenahar) [reply tests/helpers.py]: done

**Comment by ebenahar** [comment]
(ocs-venv) [ebenahar@localhost ocs-ci]$ python -m run_ocsci --cluster-name ebenahar-ocs-deployment --cluster-path ~/ebenahar-ocs/ tests/manage/test_pvc_deletion_during_io.py --capture=no
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.7.2, pytest-4.5.0, py-1.8.0, pluggy-0.12.0 -- /home/ebenahar/git/ocs-ci/ocs-venv/bin/python
cachedir: .pytest_cache
rootdir: /home/ebenahar/git/ocs-ci, inifile: pytest.ini
plugins: ocs-ci-0.1
collected 1 item                                                                                                                                                                                                  

tests/manage/test_pvc_deletion_during_io.py::TestCaseOCS371::test_run_io_and_delete_pvc 
------------------------------------------------------------------------------------------------- live log setup --------------------------------------------------------------------------------------------------
03:56:45 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig get Pod --selector=app=rook-ceph-tools -o yaml
03:56:46 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig rsh rook-ceph-tools-76c7d559b6-mfhm2 ceph auth get-key client.admin --format json-pretty
03:56:49 - MainThread - resources.ocs - INFO - Adding Secret with name secret-test-0803564532
03:56:49 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig create -f /tmp/Secretgxbngw2u -o yaml
03:56:50 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig get Secret secret-test-0803564532 -o yaml
03:56:51 - MainThread - resources.ocs - INFO - Adding CephBlockPool with name cbp-test-0803565155
03:56:51 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig create -f /tmp/CephBlockPooljji1nb2a -o yaml
03:56:52 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig get CephBlockPool cbp-test-0803565155 -o yaml
03:56:53 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig get Pod --selector=app=rook-ceph-tools -o yaml
03:56:54 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig rsh rook-ceph-tools-76c7d559b6-mfhm2 ceph osd lspools --format json-pretty
03:56:57 - MainThread - tests.helpers - INFO - Verifying that block pool cbp-test-0803565155
03:56:57 - MainThread - resources.ocs - INFO - Adding StorageClass with name storageclass-test-0803565771
03:56:57 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig create -f /tmp/StorageClassbe0pa3_v -o yaml
03:56:58 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig get StorageClass storageclass-test-0803565771 -o yaml
03:56:59 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name pvc-test-0803565956
03:56:59 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig create -f /tmp/PersistentVolumeClaimhqg5tfc0 -o yaml
03:57:00 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig get PersistentVolumeClaim pvc-test-0803565956 -o yaml
03:57:01 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig get PersistentVolumeClaim pvc-test-0803565956 -o yaml
03:57:05 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig get PersistentVolumeClaim pvc-test-0803565956 -o yaml
03:57:06 - MainThread - resources.ocs - INFO - Adding Pod with name pod-test-0803570650
03:57:06 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig create -f /tmp/POD_v4jtbo8b -o yaml
03:57:07 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig get Pod pod-test-0803570650 -o yaml
03:57:08 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig get Pod pod-test-0803570650 -o yaml
03:57:12 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig get Pod pod-test-0803570650 -o yaml
03:57:16 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig get Pod pod-test-0803570650 -o yaml
03:57:20 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig get Pod pod-test-0803570650 -o yaml
03:57:24 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig get Pod pod-test-0803570650 -o yaml
-------------------------------------------------------------------------------------------------- live log call --------------------------------------------------------------------------------------------------
03:57:25 - Thread-1 - tests.helpers - INFO - Running I/O on pod pod-test-0803570650
03:57:25 - Thread-1 - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig rsh pod-test-0803570650 bash -c "echo 123 > /var/lib/www/html/test; while true; do cp /var/lib/www/html/test /var/lib/www/html/test1; done"
03:57:30 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig delete PersistentVolumeClaim pvc-test-0803565956 --wait=false
03:57:31 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig get PersistentVolumeClaim pvc-test-0803565956
03:57:52 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig delete Pod pod-test-0803570650
03:57:53 - Thread-1 - utility.utils - ERROR - CMD error:: command terminated with exit code 137

Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib64/python3.7/threading.py", line 917, in _bootstrap_inner
    self.run()
  File "/usr/lib64/python3.7/threading.py", line 865, in run
    self._target(*self._args, **self._kwargs)
  File "/home/ebenahar/git/ocs-ci/tests/helpers.py", line 105, in run_io
    f"bash -c \"echo 123 > /var/lib/www/html/test; while true; "
  File "/home/ebenahar/git/ocs-ci/resources/pod.py", line 84, in exec_cmd_on_pod
    return self.ocp.exec_oc_cmd(rsh_cmd)
  File "/home/ebenahar/git/ocs-ci/ocs/ocp.py", line 65, in exec_oc_cmd
    out = run_cmd(cmd=oc_cmd)
  File "/home/ebenahar/git/ocs-ci/utility/utils.py", line 655, in run_cmd
    f"Error during execution of command: {cmd}."
ocs.exceptions.CommandFailed: Error during execution of command: ['oc', '-n', 'openshift-storage', '--kubeconfig', '/home/ebenahar/ebenahar-ocs/auth/kubeconfig', 'rsh', 'pod-test-0803570650', 'bash', '-c', 'echo 123 > /var/lib/www/html/test; while true; do cp /var/lib/www/html/test /var/lib/www/html/test1; done'].
Error is command terminated with exit code 137


03:57:57 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig get PersistentVolumeClaim pvc-test-0803565956
03:57:58 - MainThread - utility.utils - ERROR - CMD error:: Error from server (NotFound): persistentvolumeclaims "pvc-test-0803565956" not found

PASSED
------------------------------------------------------------------------------------------------ live log teardown ------------------------------------------------------------------------------------------------
03:57:58 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig get StorageClass storageclass-test-0803565771 -o yaml
03:57:59 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig delete StorageClass storageclass-test-0803565771
03:58:00 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig get CephBlockPool cbp-test-0803565155 -o yaml
03:58:01 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig delete CephBlockPool cbp-test-0803565155
03:58:03 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig get Secret secret-test-0803564532 -o yaml
03:58:04 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs/auth/kubeconfig delete Secret secret-test-0803564532

File: tests/helpers.py [hunk 20]
```diff
@@ -20,4 +40,296 @@ def create_unique_resource_name(resource_description, resource_type):
...
+            which the PVC belongs to
+
+    Raises:
+        AssertionError: In case of any failure
+    """
+    # as a W/A for BZ1715627, deleting PVC with wait=False
+    delete_resource(PVC, pvc_name, project_name, wait=False)
+
+
+def run_io(pod_name, project_name):
```
Comment (RazTamir) [inline tests/helpers.py]: Please move to pod class

File: tests/manage/test_pvc.py [hunk 21]
```diff
@@ -0,0 +1,43 @@
...
+    https://polarion.engineering.redhat.com/polarion/#/project/
+    OpenShiftContainerStorage/workitem?id=OCS-371
+    """
+
+    @tier1
+    def test_run_io_and_delete_pvc(self):
+        """
+        Delete PVC while IO is in progress
+        """
+        thread = Thread(
```
Comment (RazTamir) [inline tests/manage/test_pvc.py]: I think the implementation of the async of io should be done in the func itself with async parameter to the function

**Comment by ebenahar** [comment]
@RazTamir the code you've reviewed is not the latest commit. There are a lot of changes there, not sure why it wasn't uploaded, checking

**Comment by ebenahar** [comment]
@RazTamir https://github.com/red-hat-storage/ocs-ci/pull/175 will replace this PR.

**Comment by ebenahar** [comment]
@RazTamir will address your recent comments in https://github.com/red-hat-storage/ocs-ci/pull/175

File: ocs/ocp.py [hunk 22]
```diff
@@ -209,6 +211,8 @@ def wait_for_resource(
         for sample in TimeoutSampler(
             timeout, sleep, self.get, resource_name, True, selector
         ):
+            if to_delete and not sample:
+                return True
```
[the remark this answers is not in the document]
Comment (ebenahar) [reply ocs/ocp.py]: reverted tha change

File: ocs/ocp.py [hunk 23]
```diff
@@ -146,8 +146,10 @@ def delete(self, yaml_file=None, resource_name='', wait=True):
             command += f"{self.kind} {resource_name}"
         else:
             command += f"-f {yaml_file}"
-        if wait:
-            command += " --wait=true"
+
+        # default value for wait is true
```
[the remark this answers is not in the document]
Comment (ebenahar) [reply ocs/ocp.py]: Right. Added it because it was my assumption before I discovered the real cause - https://bugzilla.redhat.com/show_bug.cgi?id=1715627
And forgot to change it back. Will revert

File: tests/manage/test_pvc.py (hunk 21 above)
Comment (ebenahar) [reply tests/manage/test_pvc.py]: done

File: tests/helpers.py (hunk 20 above)
Comment (ebenahar) [reply tests/helpers.py]: done

File: ocs/ocp.py (hunk 23 above)
[the remark this answers is not in the document]
Comment (ebenahar) [reply ocs/ocp.py]: @RazTamir actually, it seems that the default is true..I see this when trying to delete a PVC while it is attached to a pod and its status is Terminating (the pod has to be deleted). The deletion never finishes and gets stuck.
